app/policies/pro_membership_policy.rb

```diff
@@ -0,0 +1,5 @@
+class ProMembershipPolicy < ApplicationPolicy
+  def create?
+    !(user.pro? || user&.pro_membership&.expired?)
```

Comment:
It's not 100% clear to me when user should be able to create a pro membership. Currently, if a user has an expired membeship, this method will return `false` and they won't be able to create a pro membership. It seems not correct. I would write the logic as smth like this: `!user.pro && (user.pro_membership.nil? || user.pro_membership.expired?)` , or are my assumptions incorrect?

Reply:
A pro membership is a one time thing. You create it once and then either renew it or let it expire. But I agree, the logic is convoluted. I can just check if the `pro_membership` is nil or not :)

Reply:
That makes sense, I thought of it to be one-time, but looked for a reason of the existing logic :)